utils/mgmt_system/google.py

```diff
@@ -0,0 +1,290 @@
+# coding: utf-8
+"""Backend management system classes
+
+Used to communicate with providers without using CFME facilities
+"""
+
+from oauth2client.client import flow_from_clientsecrets
+from oauth2client.file import Storage
+from oauth2client.tools import run_flow, argparser
+from googleapiclient.discovery import build
+from utils.mgmt_system.base import MgmtSystemAPIBase, VMInfo
+from utils.mgmt_system.exceptions import VMInstanceNotFound
+from utils.log import logger
+from utils.wait import wait_for
+
+
+class GoogleCloudSystem (MgmtSystemAPIBase):
+    """
+    Client to Google Cloud Platform API
+
+    """
+
+    states = {
+        'running': ('RUNNING',),
+        'stopped': ('TERMINATED',),
+        'starting': ('STAGING'),
+        'stopping': ('STOPPING'),
+    }
+
+    def __init__(self, project=None, zone=None, default_instance_name=None, scope=None,
+            oauth2_storage=None, client_secrets=None):
+
+        self._project = project
+        self._zone = zone
+
+        # Default name of the instance name from config file
+        self._instance_name = default_instance_name
```

Comment:
Does google cloud allow multiple instances to be named the same thing?
Should we change the name of this to ```default_instance_name```?